tests/test_filter_jobs.py

```diff
@@ -11,10 +11,16 @@
 def test_filter_jobs(base_url, selenium):
     """Open resultset page and filter for platform"""
     page = TreeherderPage(selenium, base_url).open()
-    platform = u'Linux'
+    all_builds = page.all_builds
 
+    platform = u'Linux'
     page.filter_by(platform)
-    assert platform in page.result_sets[0].builds[0].platform_name
+    filtered_builds = page.all_builds
+
+    buildname = page.result_sets[0].builds[0].platform_name
+    assert platform in buildname
+
+    assert all_builds > filtered_builds
```

Comment:
This needs to check the length of the build lists, so it should be `assert len(all_builds) > len(filtered_builds)`. When trying this locally I get failures either because the builds are still loading (so all_builds is less than filtered_builds) or because both lists are the same length. This is because we're returning all builds regardless of whether they are displayed.

Unfortunately there's no handy 'filter-shown' class like there is for jobs, but one potential solution to this is to filter the list in the `builds` property of `ResultSet`, such as:

```python
class ResultSet(Region):
 
    _platform_locator = (By.CLASS_NAME, 'platform')
 
    @property
    def builds(self):
        return [self.Build(self.page, root=el) for el in self.find_elements(*self._platform_locator) if el.is_displayed]
```